stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -805,14 +808,45 @@ public func _graph<State : _TensorArrayProtocolEnhanced,
                                             outputs: outputTensorHandles)
 
     debugLog("Creating output model instance.")
-    let newState = state._makeInstance(owning: returnValues.dropLast(
-                                         Data._typeList.count))
-    let result = Result(
-      _copying: returnValues.dropFirst(Int(state._tensorHandleCount)))
+    let newState = state._makeInstance(owning: returnValues.prefix(
+                                         Int(state._tensorHandleCount)))
+    let resultValues = returnValues.dropFirst(Int(state._tensorHandleCount))
+    let result = resultValues.isEmpty ? nil : Result(_copying: resultValues)
     return (newState, result)
   }
 }
 
+public func _graph<State : _TensorArrayProtocolEnhanced,
+                   Data : TensorGroup,
+                   Result : TensorGroup>(
+  with state: State,
+  in fn: @escaping (State, Data) -> (State, Result)
+) -> (State, Data) -> (State, Result) {
+  let graphFunction = _graphInternal(with: state, in: fn)
+  return { (state: State, data: Data) in
+    let result = graphFunction(state, data)
+    return (result.0, result.1!)
+  }
+}
+
+public func _graph<State : _TensorArrayProtocolEnhanced,
+                   Data : TensorGroup>(
+  with state: State,
+  in fn: @escaping (State, Data) -> (State)
```

Comment:
Remove parentheses around the return type.
```suggestion
  in fn: @escaping (State, Data) -> State
```